polaris-core/src/main/java/org/apache/polaris/core/persistence/resolver/ResolutionManifestFactoryImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.persistence.resolver;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import jakarta.ws.rs.core.SecurityContext;
+import org.apache.polaris.core.context.CallContext;
+
+public class ResolutionManifestFactoryImpl implements ResolutionManifestFactory {
+
+  private final ResolverFactory resolverFactory;
+
+  public ResolutionManifestFactoryImpl(@Nonnull ResolverFactory resolverFactory) {
+    this.resolverFactory = resolverFactory;
+  }
+
+  @Nonnull
+  @Override
+  public PolarisResolutionManifest createResolutionManifest(
+      @Nonnull CallContext callContext,
+      @Nonnull SecurityContext securityContext,
+      @Nullable String referenceCatalogName) {
+    return new PolarisResolutionManifest(
+        callContext, resolverFactory, securityContext, referenceCatalogName);
```

Comment:
do we need the `ResolutionManifestFactory` at all or should we use call this constructor directly instead  everywhere?
callers will still need a reference to `resolverFactory`.

Reply:
In current PR the factory does look unnecessary. However, from my POV the ultimate goal of CDI-related refactorings is to get rid of `CallContext` in method _parameters_... In that case the factory may still be valuable as a request-scoped producer for on-demand (not injected) `PolarisResolutionManifest` instances.